src/integration-tests/workload/deploy_workload_test.go

```diff
@@ -3,60 +3,112 @@ package workload_test
 import (
 	"fmt"
 	"net/http"
+	"regexp"
 	"time"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	"github.com/onsi/gomega/gexec"
 )
 
-var _ = Describe("Deploy workload", func() {
+var ipAddressError = "No IP address found for service"
+
+func getServiceIP() string {
+	timeout := time.After(60 * time.Second)
+	tick := time.Tick(500 * time.Millisecond)
+
+	numBlock := "(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])"
+	validIP := regexp.MustCompile(numBlock + "\\." + numBlock + "\\." + numBlock + "\\." + numBlock)
+
+	for {
+		select {
+		case <-timeout:
+			return ipAddressError
+		case <-tick:
+			getServiceIp := runner.RunKubectlCommand("get", "service", "nginx", "-o", "jsonpath='{.status.loadBalancer.ingress[0].ip}'")
+			Eventually(getServiceIp, "5s").Should(gexec.Exit())
+			serviceIP := string(getServiceIp.Out.Contents())
+			// Remove quotes from the IP address
+			serviceIP = serviceIP[1 : len(serviceIP)-1]
+			if validIP.MatchString(serviceIP) {
+				return serviceIP
+			}
+		}
+	}
+}
 
-	It("exposes routes via GCP LBs", func() {
+var _ = Describe("Deploy workload", func() {
 
-		appUrl := fmt.Sprintf("http://%s:%s", workerAddress, nodePort)
+	It("exposes routes via LBs", func() {
+
+		if (iaas == "gcp") {
```

Comment:
Maybe we should create another test for GCP LB, instead of conditionally running tests differently, as we are already conditionally doing things in `run-k8s-integration-tests.sh` for `$iaas == gcp`
@mkjelland Thoughts?